reports/namespace-usage/main.go

```diff
@@ -0,0 +1,231 @@
+	}
+
+	nsUsedMap := make(map[string]NamespaceResource, 0)
+
+	for _, podMetrics := range podMetricsList {
+		used, namespace := GetPodUsageDetails(podMetrics)
+		list := nsUsedMap[namespace]
+		if _, exist := nsUsedMap[namespace]; exist {
+			list.addNamespaceResource(used)
+			nsUsedMap[namespace] = list
+		} else {
+			nsUsedMap[namespace] = used
+		}
+	}
+
+	// get namespace quota of namespaces to find hard limits of pods from the cluster
+	rsQuotasList, err := namespace.GetAllResourceQuotasFromCluster(kclientset)
+	if err != nil {
+		log.Fatalln("error in getting all resourcequota from cluster", err.Error())
+	}
+
+	nsQuotaMap := make(map[string]NamespaceResource, 0)
+
+	for _, rsQuota := range rsQuotasList {
+		hardLimits, namespace, err := GetPodHardLimits(rsQuota)
+		if err != nil {
+			log.Fatalln(err.Error())
+		}
+		nsQuotaMap[namespace] = hardLimits
+	}
+
+	var usageReports []UsageReport
+	// Build the total usageReport
+	for _, ns := range nsList {
+		var usageReport UsageReport
+		usageReport.Name = ns.Name
+		usageReport.Requested = nsReqMap[ns.Name]
+		usageReport.Used = nsUsedMap[ns.Name]
+		usageReport.Hardlimits = nsQuotaMap[ns.Name]
+		usageReport.ContainerCount = containerMap[ns.Name]
+		usageReports = append(usageReports, usageReport)
+	}
+
+	jsonToPost, err := BuildJsonMap(usageReports)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	// Post json to hoowdaw api
+	err = hoodaw.PostToApi(jsonToPost, hoodawApiKey, &endPoint)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+}
+
+// GetPodResourceDetails takes a Pod of type v1.Pod and collect
+// all resources summed up for all containers of the pod and return the result
+func GetPodResourceDetails(pod v1.Pod) (r NamespaceResource, namespace string, containerCount int) {
+	reqs, _ := v1.ResourceList{}, v1.ResourceList{}
+	for _, container := range pod.Spec.Containers {
+		addResourceList(reqs, container.Resources.Requests)
+		containerCount++
+	}
+	cpuReq, memoryReq := reqs[v1.ResourceCPU], reqs[v1.ResourceMemory]
+
+	r.CPU = float64(cpuReq.MilliValue())
+	r.Memory = float64(memoryReq.Value() / 1048576)
+	namespace = pod.Namespace
+	return
+}
+
+// GetPodResourceDetails takes a Pod of type v1.Pod and collect
+// all resources summed up for all containers of the pod and return the result
+func GetPodUsageDetails(podMetrics v1beta1.PodMetrics) (u NamespaceResource, namespace string) {
+	usage := v1.ResourceList{}
+	for _, container := range podMetrics.Containers {
+		addResourceList(usage, container.Usage)
+	}
+	cpuUsage, memoryUsage := usage[v1.ResourceCPU], usage[v1.ResourceMemory]
+	u.CPU = float64(cpuUsage.MilliValue())
+	u.Memory = float64(memoryUsage.Value() / 1048576)
+	namespace = podMetrics.Namespace
+	return
+}
+
+// GetPodResourceDetails takes a Pod of type v1.Pod and collect
+// all resources summed up for all containers of the pod and return the result
+func GetPodHardLimits(resourceQuota v1.ResourceQuota) (h NamespaceResource, namespace string, err error) {
```

Comment:
I can only see one package in your module here. Is there any reason these are public (exported) functions?

Reply:
Good Point. Not in particular, but i first thought this could be useful for parsing any pod data. I will make it local to this package and we can change if we need to later.